src/GraphQL/MuckTom.hs

```diff
@@ -0,0 +1,209 @@
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module GraphQL.MuckTom where
+
+-- TODO (probably incomplete, the spec is large)
+-- * input objects - I'm not super clear from the spec on how
+--   they differ from normal objects.
+-- * "extend type X" is used in examples in the spec but it's not
+--   explained anywhere?
+-- * Directives (https://facebook.github.io/graphql/#sec-Type-System.Directives)
```

Comment:
I think this happens at a different level to this code (that said I don't know what this code is)